rust-common/src/models.rs

```diff
@@ -266,6 +266,48 @@ impl DecisionModel for MemoryMappableMultiCore {
     }
 }
 
+/// A decision model capturing the memory mappable platform abstraction.
+///
+/// This type of platform is what one would expect from most COTS platforms
+/// and hardware designs, which completely or partially follows a von neumman
+/// architecture. This means that the storage elements store both data and instructions
+/// and the processors access them going through the communication elements; the latter
+/// that form the 'interconnect'. In addition to standard software processing elements,
+/// this decision model also includes programmable logic capacities on the platform.
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq, JsonSchema)]
+pub struct MM_MCoreAndPL {
+    pub processing_elems: HashSet<String>,
+    pub pl_module_available_areas: HashMap<String, u32>,
+    pub storage_elems: HashSet<String>,
+    pub communication_elems: HashSet<String>,
+    pub topology_srcs: Vec<String>,
+    pub topology_dsts: Vec<String>,
+    pub processors_frequency: HashMap<String, u64>,
+    pub processors_provisions: HashMap<String, HashMap<String, HashMap<String, f64>>>,
+    pub storage_sizes: HashMap<String, u64>,
+    pub communication_elements_max_channels: HashMap<String, u32>,
+    pub communication_elements_bit_per_sec_per_channel: HashMap<String, f64>,
+    pub pre_computed_paths: HashMap<String, HashMap<String, Vec<String>>>,
+}
+
+impl DecisionModel for MM_MCoreAndPL {
+    impl_decision_model_standard_parts!(MM_MCoreAndPL);
+
+    fn part(&self) -> HashSet<String> {

```

Comment:
Do I need to update this function somehow for my new property? There is no 1:1 mapping between the struct fields and what this function returns so I couldn't draw any conclusion really.